### PR TITLE
Add TextWatcher in Textfield

### PR DESCRIPTION
**Pull request by jperezgonzal**
## Description
Se agregó TextWatcher en el TextField y TextArea. Era requerido por algunos equipos para poder tener un listener cuando el TextField cambiara.

## Checks
Are you sure you followed all those steps? In such case, let's say with me "I solemnly declare that ...":
- [X] I have coded an example inside the Demo App,
- [ ] I have added proper tests,
- [X] I have modified the [Changelog](https://github.com/mercadolibre/fury_andesui-android/blob/develop/CHANGELOG.md) file,
- [ ] I have updated GitHub wiki,
- [ ] I have the explicit approval of one or more members of the UX Team,
- [ ] I have updated the version in gradle.properties file.

## Change type
This is easy, let us know what this code is about:
- [ ] This code adds a new component
- [X] This code includes improvements to an existent component
- [ ] This code improves or modifies ONLY the Demo App.

## Check common errors
Whether you are an Android or iOS developer and if you're still there then we'll give you a present:
https://proandroiddev.com/how-to-maximize-androids-ui-reusability-5-common-mistakes-cb2571216a9f

### CODE REVIEW

**Comment by furyGorgoryAdmin** [comment]
### Informative Menu <h2>  ### 

### Use any of these tags in a pull request description or leave a comment ### 
| Tag | Description |
|---|---|
| **[canary]**| Creates a build that can only be deployed with canary strategy. | 
| **[build]** |  Creates version or build from a branch that has been tagged as "no releasable". | 
| **[name:version-name]** |Specify the name of the build. For example: [build] [name:1.2.3-name] or [canary] [name:1.2.3-name].|
| **[major]** | To automatically increase the first position of the version. For example: [build] [major] or [canary] [major].|
| **[minor]** | To automatically increase the second position of the version. For example: [build] [minor] or [canary] [minor]. |
| **[patch]** | To automatically increase the third position of the version. For example: [build] [patch] or [canary] [patch].|  

[For more information about ci-cd](http://docs.furycloud.io/?lang=esp&section=businessassurance)
[For more information about semver](https://semver.org/#summary)

File: components/src/main/java/com/mercadolibre/android/andesui/textfield/AndesTextarea.kt [hunk 1]
```diff
@@ -212,6 +223,15 @@ class AndesTextarea : ConstraintLayout {
         textComponent.typeface = config.typeface
     }
 
+    /**
+     * Set the TextWatcher of the edit text.
+     */
+    private fun setupTextWatcher() {
+        if (andesTextareaAttrs.textWatcher != null) {
+            textComponent.addTextChangedListener(andesTextareaAttrs.textWatcher)
+        }
```
Comment (jiurchuk) [inline components/src/main/java/com/mercadolibre/android/andesui/textfield/AndesTextarea.kt:232]:
```suggestion
        andesTextareaAttrs.textWatcher? 
            textComponent.addTextChangedListener(andesTextareaAttrs.textWatcher)
```

File: components/src/main/java/com/mercadolibre/android/andesui/textfield/AndesTextfield.kt [hunk 2]
```diff
@@ -262,6 +273,15 @@ class AndesTextfield : ConstraintLayout {
         textComponent.inputType = inputType
     }
 
+    /**
+     * Set the TextWatcher of the edit text.
+     */
+    private fun setupTextWatcher() {
+        if (andesTextfieldAttrs.textWatcher != null) {
+            textComponent.addTextChangedListener(andesTextfieldAttrs.textWatcher)
+        }
```
Comment (jiurchuk) [inline components/src/main/java/com/mercadolibre/android/andesui/textfield/AndesTextfield.kt:282]:
```suggestion
        andesTextfieldAttrs.textWatcher?
            textComponent.addTextChangedListener(andesTextfieldAttrs.textWatcher)
```

File: demoapp/src/main/java/com/mercadolibre/android/andesui/demoapp/feature/TextfieldShowcaseActivity.kt [hunk 3]
```diff
@@ -20,6 +22,7 @@ import com.mercadolibre.android.andesui.textfield.content.AndesTextfieldLeftCont
 import com.mercadolibre.android.andesui.textfield.content.AndesTextfieldRightContent
 import com.mercadolibre.android.andesui.textfield.state.AndesTextfieldState
 
+
```
Comment (jiurchuk) [inline demoapp/src/main/java/com/mercadolibre/android/andesui/demoapp/feature/TextfieldShowcaseActivity.kt:25]: Empty line

File: demoapp/src/main/java/com/mercadolibre/android/andesui/demoapp/feature/TextfieldShowcaseActivity.kt [hunk 4]
```diff
@@ -213,6 +216,19 @@ class TextfieldShowcaseActivity : AppCompatActivity() {
...
+                override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {
+                    Toast.makeText(context, "Text changed: $s", Toast.LENGTH_SHORT).show()
+                }
+
+                override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {
+                    // TODO Auto-generated method stub
+                }
+
+                override fun afterTextChanged(s: Editable) {
+                    // TODO Auto-generated method stub
```
Comment (jiurchuk) [inline demoapp/src/main/java/com/mercadolibre/android/andesui/demoapp/feature/TextfieldShowcaseActivity.kt:229]:
```suggestion
                    //no-op
```

File: demoapp/src/main/java/com/mercadolibre/android/andesui/demoapp/feature/TextfieldShowcaseActivity.kt [hunk 5]
```diff
@@ -213,6 +216,19 @@ class TextfieldShowcaseActivity : AppCompatActivity() {
             // Set action clear
             val textfield1 = layoutTextfield.findViewById<AndesTextfield>(R.id.andesTextfield1)
             textfield1.text = context.resources.getString(R.string.andesui_demoapp_textfield_placeholder)
+            textfield1.textWatcher = object : TextWatcher {
+                override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {
+                    Toast.makeText(context, "Text changed: $s", Toast.LENGTH_SHORT).show()
+                }
+
+                override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {
+                    // TODO Auto-generated method stub
```
Comment (jiurchuk) [inline demoapp/src/main/java/com/mercadolibre/android/andesui/demoapp/feature/TextfieldShowcaseActivity.kt:225]:
```suggestion
                    //no-op
```

**Comment by RodriP** [comment]
Podemos meter algun test de este nuevo listener?

**Pull request merged**